docs/sampling/README.md

```diff
@@ -0,0 +1,282 @@
+and Logs-to-Metrics pipelines, which accurately count Spans and Log
+Records of telemetry, before sampling, in a probabilistic sense.
+
+These attributes MAY be modified by components in a collection
+pipeline to convey successive sampling that has been carried out for a
+particular item of telemetry, using the conventions for consistent
+sampling described here.  In that sense, telemetry consumers
+should see these attributes as telemetry metadata.
+
+## Probability sampling
+
+The OpenTelemetry sampling decision is defined in terms of a Threshold
+value and a Randomness value, each containing 56 bits of information.
+
+A constant known as _maximum adjusted count_ (`MaxAdjustedCount`),
+with value `0x100000000000000`, (which can also be expressed
+`0x1p+56`, `math.Pow(2, 56)`, or `math.Ldexp(1, 56)`), defines the
+exclusive upper limit of these values.
+
+Logically, both Threshold (`T`) and Randomness (`R`) are represented
+as unsigned integers in the range `0` through `0xffffffffffffff` or
+`MaxAdjustedCount - 1`.  Items of telemetry are selected (i.e.,
+"sampled") when their Threshold value is less than or equal to their
+Randomness value, or `T <= R`.
+
+Sampling probability is defined by the following expression:
+
+```
+Probability = (MaxAdjustedCount - Threshold) / MaxAdjustedCount
+```
+
+In a Span-to-Metrics or Logs-to-Metrics pipeline, each item of
+telemetry is representative of an _adjusted count_ number of items in
+the original population.  Adjusted count is the inverse of sampling
+probability, and `MaxAdjustedCount` (defined above) is the inverse of
+the smallest supported sampling probability (which can also be
+represented as `0x1p-56`, `math.Pow(2, -56)`, or `math.Ldexp(1,
+-56)`).
+
+For the tracing signal, Threshold and Randomness propagate via W3C
+Trace Context `tracestate`.  When they appear in the `tracestate`, the
+Threshold and Randomness properties are called "T-value" and
+"R-value"; they are represented in the OpenTelemetry section of the
+`tracestate` (having vendor tag `ot`), using properties named `th` and
+`rv`, respectively.
+
+For the logs signal, which generally does not record the W3C Trace
+Context `tracestate`, sampling attributes are meant to be expressed
+using log record attributes with the same definition as T-value and
+R-value.
+
+For more information on how to perform and interpret probability
+sampling based on these properties, [consult OTEP 235][OTEP235].
+
+### Overriding sampling decisions
+
+Instrumentation authors and end-users that wish to prioritize an item
+of telemetry for collection in spite of sampling can add a
+`sampling.priority` attribute.  This attribute is a suggestion, a way
+of recognizing the importance of a certain event and requesting
+additional consideration from the collection pipeline.  For example, a
+user could write this code snippet:
+
+```
+   if err := doSomething(); err != nil {
+     if err == VERY_SERIOUS {
+       span.SetAttribute("sampling.priority", 10000)
```

Comment:
Nit: Is the idea that even if an original sampling decision was to drop, you can override it by setting this attribute? If so, IMHO, the name "priority" doesn't directly convey the override aspect.  Hence, depending on what use cases we want to allow (change decision from drop -> keep, or change decision from keep -> drop, or both), I wonder if this should be a simple bool attribute such as "sampling.keep" or "sampling.include" or "sampling.override".

In the future, if we expect pipelines to prioritize between multiple levels (e.g., if I could keep only X%, prefer the ones with the higher priority), then we could still add a new attribute to convey that relative priority.

Thoughts?

Reply:
@kalyanaj I tried to address your concern in https://github.com/open-telemetry/semantic-conventions/pull/793#issuecomment-1985953857.

@lmolkova Yes. Another way of saying this (which OTEP 235 explains) is that Thresholds cannot decrease as a result of sampling, but they can increase. OTEP 235 writes

>  In other words, th MUST NOT be decreased (as it is not possible to retroactively adjust an earlier stage's sampling probability), and it MUST be increased if a lower sampling probability was used.

The trace SDK specification confines what we can do for SDKs, in the sense that there is not Sampler decision corresponding with not sampling, but recording the span and possibly changing the decision to export the unsampled span anyway. See e.g.,

https://github.com/open-telemetry/opentelemetry-specification/issues/2918
https://github.com/open-telemetry/opentelemetry-specification/issues/2986